internal/api/mfa.go

```diff
@@ -676,6 +874,115 @@ func (a *API) verifyPhoneFactor(w http.ResponseWriter, r *http.Request, params *
 	return sendJSON(w, http.StatusOK, token)
 }
 
+func (a *API) verifyWebAuthnFactor(w http.ResponseWriter, r *http.Request, params *VerifyFactorParams) error {
+	ctx := r.Context()
+	user := getUser(ctx)
+	factor := getFactor(ctx)
+	db := a.db.WithContext(ctx)
+
+	var webAuthn *webauthn.WebAuthn
+	var credential *webauthn.Credential
+	var err error
+
+	switch {
+	case params.WebAuthn == nil:
+		return badRequestError(ErrorCodeValidationFailed, "WebAuthn config required")
+	case factor.IsVerified() && params.WebAuthn.AssertionResponse == nil:
+		return badRequestError(ErrorCodeValidationFailed, "WebAuthn Assertion Response required to login")
+	case factor.IsUnverified() && params.WebAuthn.CreationResponse == nil:
+		return badRequestError(ErrorCodeValidationFailed, "WebAuthn Creation Response required to login")
+	default:
+		webAuthn, err = params.WebAuthn.ToConfig()
+		if err != nil {
+			return err
+		}
+	}
+
+	challenge, err := a.validateChallenge(r, db, factor, params.ChallengeID)
+	if err != nil {
+		return err
+	}
+	webAuthnSession := *challenge.WebAuthnSessionData.SessionData
+	// Once the challenge is validated, we consume the challenge
+	if err := db.Destroy(challenge); err != nil {
+		return internalServerError("Database error deleting challenge").WithInternalError(err)
+	}
+
+	if factor.IsUnverified() {
+		creationResponseJSON, err := json.Marshal(params.WebAuthn.CreationResponse)
+		if err != nil {
+			return badRequestError(ErrorCodeValidationFailed, "Failed to marshal CreationResponse to JSON")
+		}
+		creationResponseReader := bytes.NewReader(creationResponseJSON)
+		parsedResponse, err := wbnprotocol.ParseCredentialCreationResponseBody(creationResponseReader)
```

Comment:
Hmm it's because of this `Parse` method.

A trick you can do is to encode the `CredentialRequestOptions` as a `json.RawMessage` instead, and then just pass it to `ParseCredentialCreationResponseBody()` ... if it's not used elsewhere.